cargo-shuttle/src/client.rs

```diff
@@ -84,11 +85,11 @@ impl ShuttleApiClient {
             .context("parsing name check response")
     }
 
-    pub async fn deploy(
+    pub async fn deploy<T: DeserializeOwned>(
         &self,
         project: &str,
         deployment_req: DeploymentRequest,
-    ) -> Result<deployment::Response> {
+    ) -> Result<T> {
```

Comment:
suggestion: I would really just have two functions at this point. Ideally we would have two client, one for each version, even though that's a big refactor. The issue here is that it makes the function more difficult to understand (what's the expected output ? It varies between call site), and a change at the call site could make the function fail in non-obvious way.

Reply:
Split this in two functions (beta & alpha) in the same client for now. I suggest to create a client for each platform later on, after we clarify which is the differentiating logic for all commands.